synapse/config/emailconfig.py

```diff
@@ -74,19 +74,33 @@ def read_config(self, config, **kwargs):
             "renew_at"
         )
 
-        email_trust_identity_server_for_password_resets = email_config.get(
-            "trust_identity_server_for_password_resets", False
+        self.email_threepid_behaviour = (
+            # Have Synapse handle the email sending if account_threepid_delegate
+            # is not defined
+            "remote"
+            if self.account_threepid_delegate
+            else "local"
         )
-        self.email_password_reset_behaviour = (
-            "remote" if email_trust_identity_server_for_password_resets else "local"
-        )
-        self.password_resets_were_disabled_due_to_email_config = False
-        if self.email_password_reset_behaviour == "local" and email_config == {}:
+        # Prior to Synapse v1.4.0, there used to be another option that defined whether Synapse
+        # would use an identity server to password reset tokens on its behalf. We now warn the
+        # user if they have this set and tell them to use the updated option.
```

Comment:
This doesn't warn but kills synapse. I'd strongly prefer to avoid doing this, as otherwise we need to scream from the roof tops that the upgrade may cause your synapse to not start. Can we do something backwards compatible?

Reply:
We need to know what identity server they'd prefer to use to handle registration and password resets. We could just warn them with a message on startup without blocking Synapse start? Not exactly sure where in the codebase we could put the message though, given we can't log here.

Reply:
Can we keep the old behaviour? Of using the first entry in trusted_id_servers?

Reply:
We can, though that's another thing relying on the trusted id servers list.

Though this should also be eventually removed, so I say that could probably work.

Reply:
Though if their `trusted_third_party_id_servers` is empty and `trust_identity_server_for_password_resets` is set we'll just `ConfigError`?

Reply:
> Though if their `trusted_third_party_id_servers` is empty and `trust_identity_server_for_password_resets` is set we'll just `ConfigError`?

This is what I've done.